RecoEgamma/EgammaTools/plugins/HGCalPhotonIDValueMapProducer.cc

```diff
@@ -0,0 +1,221 @@
+// -*- C++ -*-
```

Comment:
There is a lot of duplicated code between this one and HGCalElectronIDValueMapProducer.cc - consider making a templated producer like HGCalEgammaIDValueMapProducer with a subroutine to handle object-specific floats (just the track-based ones for electrons, seemingly) that can be specialized when needed.

Reply:
Someday they may require more and different products from the event, so we expect the classes to eventually diverge from each other.

Reply:
May I ask you to share a roadmap for developments related to this feature.